core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionProvider.java

```diff
@@ -57,4 +57,37 @@ default FileObject getDirectFile( T connectionDetails, String path ) throws Kett
     return KettleVFS.getFileObject( pvfsUrl, new Variables(), getOpts( connectionDetails ) );
   }
 
+
+  /**
+   * Test the VFS connection with provided connection details .
+   *
+   *
+   * @return {@code true} if the provided rootPath is valid
+   */
+  default boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
+    return test( connectionDetails );
+  }
+
+  /**
+   * Method to resolve root path for a given VFS connection.
+   *
+   *
+   * @return resolved root path.
+   */
+
+  default String getResolvedRootPath( T connectionDetails ) {
+    return connectionDetails.getRootPath();
```

Comment:
Better to fix this implementation possibly returning an empty value, instead of null in that case.